Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,419 @@
+/* Copyright 2019-2020 Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "ParticleExtrema.H"
+#include "WarpX.H"
+#include "Utils/WarpXConst.H"
+#if (defined WARPX_QED)
+#include "Particles/ElementaryProcess/QEDInternals/QedChiFunctions.H"
+#endif
+
+#include <AMReX_REAL.H>
+#include <AMReX_ParticleReduce.H>
+
+#include <iostream>
+#include <cmath>
+#include <limits>
+
+
+using namespace amrex;
+
+// constructor
+ParticleExtrema::ParticleExtrema (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // read species name
+    ParmParse pp(rd_name);
+    pp.get("species",m_species_name);
+
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    // get number of species (int)
+    auto nSpecies = mypc.nSpecies();
+
+    // get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
+
+    // loop over species
+    for (int i_s = 0; i_s < nSpecies; ++i_s)
+    {
+        // only chosen species does
+        if (species_names[i_s] != m_species_name) { continue; }
+
+        // get WarpXParticleContainer class object
+        auto & myspc = mypc.GetParticleContainer(i_s);
+
+        if (myspc.has_breit_wheeler() || myspc.has_quantum_sync())
+        {
+            // resize data array for QED species
+            m_data.resize(18,0.0);
```

Comment:
```suggestion
            const int num_quantities = 18;
            m_data.resize(num_quantities,0.0);
```
I would suggest to write something like this to avoid unnamed "magic" numbers!